src/App.tsx

```diff
@@ -15,36 +19,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETICAL = 'alphabetical',
+  LENGTH = 'length',
+  DEFAULT = 'default',
+}
+
+interface FilterParams {
+  sortType: SortType;
+  reversed: boolean;
+}
+
+const getPreparedGoods = (
+  goods: Good[],
+  { sortType, reversed }: FilterParams,
+): Good[] => {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ALPHABETICAL:
+          return good1.localeCompare(good2);
+
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [reversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.DEFAULT);
+
+  const handleSortAlphabetically = () => setSortType(SortType.ALPHABETICAL);
+  const handleSortByLength = () => setSortType(SortType.LENGTH);
+  const handleReverse = () => setReversed(!reversed);
+  const handleReset = () => {
+    setSortType(SortType.DEFAULT);
+    setReversed(false);
+  };
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortType,
+    reversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.ALPHABETICAL,
+          })}
```

Comment:
[UX] Please mark the active button to explain to the user what action is active. In this case, when 'Sort alphabetically' is the active sorting method, the button should be visually distinct without the 'is-light' class.